iot_config/src/helium_netids.rs

```diff
@@ -0,0 +1,377 @@
+use crate::lora_field::{self, DevAddrConstraint, LoraField, NetIdField};
+use helium_proto::services::iot_config::org_create_helium_req_v1::HeliumNetId as ProtoNetId;
+use std::{collections::HashSet, ops::RangeInclusive};
+
+const TYPE_0_ID: NetIdField = LoraField(0x00003c);
+const TYPE_3_ID: NetIdField = LoraField(0x60002d);
+const TYPE_6_ID: NetIdField = LoraField(0xc00053);
+const TYPE_0_RANGE: RangeInclusive<u32> = 2_013_265_920..=2_046_820_351;
+const TYPE_3_RANGE: RangeInclusive<u32> = 3_763_994_624..=3_764_125_695;
+const TYPE_6_RANGE: RangeInclusive<u32> = 4_227_943_424..=4_227_944_447;
+
+pub enum HeliumNetId {
+    Type0,
+    Type3,
+    Type6,
+}
+
+impl HeliumNetId {
+    pub fn id(&self) -> NetIdField {
+        match *self {
+            HeliumNetId::Type0 => TYPE_0_ID,
+            HeliumNetId::Type3 => TYPE_3_ID,
+            HeliumNetId::Type6 => TYPE_6_ID,
+        }
+    }
+
+    pub fn addr_range(&self) -> RangeInclusive<u32> {
+        match *self {
+            HeliumNetId::Type0 => TYPE_0_RANGE,
+            HeliumNetId::Type3 => TYPE_3_RANGE,
+            HeliumNetId::Type6 => TYPE_6_RANGE,
+        }
+    }
+}
+
+impl TryFrom<NetIdField> for HeliumNetId {
+    type Error = &'static str;
+
+    fn try_from(field: NetIdField) -> Result<Self, Self::Error> {
+        let id = match field {
+            TYPE_0_ID => HeliumNetId::Type0,
+            TYPE_3_ID => HeliumNetId::Type3,
+            TYPE_6_ID => HeliumNetId::Type6,
+            _ => return Err("not a helium id"),
+        };
+        Ok(id)
+    }
+}
+
+#[async_trait::async_trait]
+pub trait AddressStore {
+    type Error;
+
+    async fn get_used_addrs(&mut self) -> Result<Vec<u32>, Self::Error>;
+    async fn claim_addrs(&mut self, new_addrs: &[u32]) -> Result<(), Self::Error>;
+    async fn release_addrs(&mut self, released_addrs: &[u32]) -> Result<(), Self::Error>;
+}
+
+#[async_trait::async_trait]
+impl AddressStore for sqlx::Pool<sqlx::Postgres> {
```

Comment:
Nothing appears to be using the Pool impl.

Reply:
indeed; i could have sworn i deleted that